druntime/test/shared/Makefile

```diff
@@ -1,34 +1,38 @@
-# druntime DLL only partially supported on Windows
-LINK_SHARED:=$(if $(findstring $(OS),windows),$(SHARED),1)
+# SHARED (from druntime/Makefile) is `1` for platforms supporting a shared druntime library, otherwise empty
+LINK_SHARED:=$(SHARED)
 
 include ../common.mak # affected by LINK_SHARED!
 
+ifneq (,$(LINK_SHARED))
+    # TODO: enable tests on Windows
+    ifeq (windows,$(OS))
+        TESTS:=link linkD linkDR loadDR
+    else
+        TESTS:=link load linkD linkDR loadDR host finalize dynamiccast \
+               link_linkdep load_linkdep link_loaddep load_loaddep load_13414
+    endif
+endif
+# there are extra tests for Windows, not requiring a druntime DLL
+ifeq (windows,$(OS))
+    TESTS+=loadlibwin dllrefcount dllgc dynamiccast
+endif
+
+DOTIMPLIB:=$(if $(findstring $(OS),windows),.lib,$(DOTDLL))
+
 .PHONY: all clean
+all: $(addprefix $(ROOT)/,$(addsuffix .done,$(TESTS)))
 
-ifeq (windows,$(OS))
+ifeq (windows,$(OS)) # extra tests on Windows
+
+# dmd -shared does not (yet) imply -visibility=public
+$(ROOT)/%$(DOTDLL): DFLAGS += -visibility=public
 
-TESTS:=loadlibwin dllrefcount dllgc dynamiccast
 ifeq ($(SHARED),1)
 TESTS+=link
```

Comment:
I think this is superfluous now, already added in line 9 (`LINK_SHARED=SHARED`). And adding to TESTS potentially too late here anyway. [line re-added after rebase]

Reply:
Also re-added: prepending to PATH in line 33. I was pretty sure this wouldn't have any effect without `export`.

Reply:
Then there's a new Windows-specific `SimpleDllMain` mixin in `druntime/test/shared/src/lib.d` in master; ideally that'd be DMD-specific, to make clear that the other compilers don't need such extra stuff on Windows.

Thanks Rainer!

Reply:
> Also re-added: prepending to PATH in line 33. I was pretty sure this wouldn't have any effect without export.

Oh well, looks like that somehow works without `export` and is needed.